mycroft/skills/skill_manager.py

```diff
@@ -451,33 +492,45 @@ def deactivate_skill(self, message):
                 if message.data['skill'] == skill_loader.skill_id:
                     LOG.info("Deactivating skill: " + skill_loader.skill_id)
                     skill_loader.deactivate()
+            for skill_loader in self.plugin_skills.values():
+                if message.data['skill'] == skill_loader.skill_id:
+                    skill_loader.deactivate()
+            # TODO handle external skills, OVOSAbstractApp/Hivemind skills are not accounted for
         except Exception:
             LOG.exception('Failed to deactivate ' + message.data['skill'])
 
     def deactivate_except(self, message):
         """Deactivate all skills except the provided."""
         try:
             skill_to_keep = message.data['skill']
-            LOG.info('Deactivating all skills except {}'.format(skill_to_keep))
-            loaded_skill_file_names = [
-                os.path.basename(skill_dir) for skill_dir in self.skill_loaders
-            ]
-            if skill_to_keep in loaded_skill_file_names:
-                for skill in self.skill_loaders.values():
-                    if skill.skill_id != skill_to_keep:
-                        skill.deactivate()
-            else:
-                LOG.info('Couldn\'t find skill ' + message.data['skill'])
+            LOG.info(f'Deactivating all skills except {skill_to_keep}')
+            for skill in self.skill_loaders.values():
```

Comment:
same as above; maybe it would be useful to define something like:

```python
@property
def all_loaded_skills(self):
    return {**self.skill_loaders, **self.plugin_skills}
    # TODO: Handle external skills....
```